internal/services/machinelearning/machine_learning_registry_resource.go

```diff
@@ -0,0 +1,449 @@
+				}
+
+				param.Properties.RegionDetails = &regions
+			}
+
+			response, err := client.RegistriesCreateOrUpdate(ctx, id, *param)
+			if err != nil && (response.HttpResponse == nil || response.HttpResponse.StatusCode != 202) {
+				return fmt.Errorf("creating %s: %+v", id, err)
+			}
+			if response.HttpResponse != nil {
+				pollerType, err := custompollers.NewMachineLearningRegistryPoller(client, response.HttpResponse)
+				if err != nil {
+					return fmt.Errorf("creating poller: %+v", err)
+				}
+				if pollerType == nil {
+					return fmt.Errorf("no poller created for updating %s", id)
+				}
+				poller := pollers.NewPoller(pollerType, 10*time.Second, pollers.DefaultNumberOfDroppedConnectionsToAllow)
+				if err := poller.PollUntilDone(ctx); err != nil {
+					return fmt.Errorf("polling update of %s: %+v", id, err)
+				}
+			}
+
+			metadata.SetID(id)
+			return nil
+		},
+	}
+}
+
+func (r MachineLearningRegistry) Delete() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.MachineLearning.RegistryManagement
+
+			id, err := registrymanagement.ParseRegistryID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			if err := client.RegistriesDeleteThenPoll(ctx, *id); err != nil {
+				return fmt.Errorf("deleting %s: %+v", *id, err)
+			}
+
+			return nil
+		},
+	}
+}
+
+func expandRegistryRegionDetail(input ReplicationRegion) registrymanagement.RegistryRegionArmDetails {
+	return registrymanagement.RegistryRegionArmDetails{
+		Location: pointer.To(input.Location),
+		AcrDetails: &[]registrymanagement.AcrDetails{
+			{
+				SystemCreatedAcrAccount: &registrymanagement.SystemCreatedAcrAccount{
+					AcrAccountSku: pointer.To(string(registrymanagement.SkuTierPremium)),
+				},
+			},
+		},
+		StorageAccountDetails: &[]registrymanagement.StorageAccountDetails{
+			{
+				SystemCreatedStorageAccount: &registrymanagement.SystemCreatedStorageAccount{
+					StorageAccountHnsEnabled: pointer.To(input.HnsEnabled),
+					StorageAccountType:       pointer.To(input.StorageAccountType),
+				},
+			},
+		},
+	}
+}
+
+func flattenRegistryRegionDetails(input *[]registrymanagement.RegistryRegionArmDetails) []ReplicationRegion {
+	result := make([]ReplicationRegion, 0)
+	if input == nil || len(*input) == 0 {
+		return result
+	}
+
+	for _, item := range *input {
+		var region ReplicationRegion
+		region.Location = pointer.From(item.Location)
+
+		if sa := pointer.From(item.StorageAccountDetails); len(sa) > 0 {
+			if systemAccount := sa[0].SystemCreatedStorageAccount; systemAccount != nil {
+				region.StorageAccountType = pointer.From(systemAccount.StorageAccountType)
+				region.HnsEnabled = pointer.From(systemAccount.StorageAccountHnsEnabled)
+				region.SystemCreatedStorageAccountId = pointer.From(pointer.From(systemAccount.ArmResourceId).ResourceId)
```

Comment:
as noted before, this should be nil checked and set as per the normal pattern rather than chaining